include/zephyr/drivers/gpio_keys.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2022 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_GPIO_KEYS_H_
+#define ZEPHYR_INCLUDE_DRIVERS_GPIO_KEYS_H_
+
+#include <zephyr/drivers/gpio.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief GPIO Keys Driver APIs
+ * @defgroup gpio_keys_interface GPIO KeysDriver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * This structure is common to all GPIO Keys drivers and is expected to be
+ * the first element in the object pointed to by the config field
+ * in the device structure.
+ */
+struct gpio_keys_config {
+	/** GPIO specification from devicetree */
+	struct gpio_dt_spec spec;
+	/** Debounce interval in milliseconds from devicetree */
+	uint32_t debounce_ms;
+};
+
+__subsystem struct gpio_keys_api {
+	int (*enable_interrupt)(const struct device *dev, gpio_callback_handler_t cb);
+	int (*disable_interrupt)(const struct device *dev);
+	int (*get_pin)(const struct device *dev);
+	int (*get_pin_raw)(const struct device *dev);
+};
```

Comment:
Would it be possible to reuse `struct gpio_driver_api` instead and make this look like a normal gpio device? That way one could go from the raw gpio port to the debounced one using the same apis.

Reply:
This was my initial thought on implementation.  Since gpio_keys is a layer on top of gpio, most of the API implementation would be a wrapper of passing items of `struct gpio_dt_spec` to the specific gpio api.  I wanted to keep this lightweight as possible, so I wanted to give bare minimum APIs here.  If there's a strong desire to reuse the API, I can make this change.